components/storage/StorageTable.tsx

```diff
@@ -79,25 +79,52 @@ const Table: React.FC<TableProps> = ({ services, selectedAnswers, questions }) =
   const columnHelper = createColumnHelper<TableRow>();
   const tableContainerRef = useRef<HTMLDivElement>(null);
 
-const tableData: TableRow[] = useMemo(() => {
-  const uniqueFeatureNames = new Set<string>();
-  services.forEach(service => {
+  // --- FEATURE SORTING LOGIC ---
+  const sortedFeatureNames = useMemo(() => {
+    const featureNamesFromForm = questions.map(q => q.affected_feature);
+    const uniqueServiceFeatureNames = new Set<string>();
+    services.forEach(service => {
       service.features?.forEach(feature => {
-          uniqueFeatureNames.add(feature.name);
+        uniqueServiceFeatureNames.add(feature.name);
       });
-  });
+    });
+
+    const featuresNotInForm: string[] = [];
+    const featuresInFormOrder: string[] = [];
 
-  // TODO: sort the table rows by questions first, then other features alphabetically
-  const orderedFeatureNames = Array.from(uniqueFeatureNames).sort();
+    // Separate features: those in the form, and those not
+    Array.from(uniqueServiceFeatureNames).forEach(featureName => {
+      if (featureNamesFromForm.includes(featureName)) {
+        featuresInFormOrder.push(featureName);
+      } else {
+        featuresNotInForm.push(featureName);
+      }
+    });
```

Comment:
In this part, you could use this `filter` function https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

You might only need the to filter for the ones that are not in the form?

Reply:
yes!